src/data/actions/courseInfo.js

```diff
@@ -66,15 +81,52 @@ export function fetchCourseInfo(id) {
   };
 }
 
-export function createCourse(courseData) {
+export function createCourseRun(course, courseRunData) {
+  return (dispatch) => {
+    dispatch(createNewCourseRun());
+    if (!course.key) {
+      dispatch(courseCreateRunFail('Course Run create failed, please try again or contact support. Error( Course create incomplete )'));
+      return null;
+    }
+
+    const data = {
+      course: course.key,
+      start: courseRunData.start,
+      end: courseRunData.end,
+    };
+    return DiscoveryDataApiService.createCourseRun(data)
+      .then((response) => {
+        const courseRun = response.data;
+        dispatch(courseCreateRunSuccess(courseRun));
+        return dispatch(push(`/courses/${course.uuid}/`));
+      })
+      .catch((response) => {
+        let errorMsg = `Course Run create failed, please try again or contact support. \nError ( ${response.message} )`;
+        const errorData = response.response.data;
+        if (errorData) {
+          // More options in the error response
+          errorMsg += '\n\n';
+
+          Object.keys(errorData).forEach((errorKey) => {
+            if (Object.prototype.hasOwnProperty.call(errorData, errorKey)) {
+              errorMsg += `${errorKey}: ${errorData[errorKey]}`;
```

Comment:
Maybe put a newline at the end of this for readability?

Reply:
The new line does not work because it is not respected by the HTML block.  I think we need a better way to handle these errors in general.

Reply:
Hmm - you could make an array of error messages. Then loop over them like:

```
errorMessages.map() => {
  <div>{message}</div>
}
```

But really, we want to eventually actually tie the error with the field. Maybe add that to the bucket doc and just move on for now?

Reply:
That was my intention

Reply:
I will try the div option though.